server/hwapi/external/c3/client.py

```diff
@@ -152,3 +180,63 @@ def _load_certified_configurations_from_response(
             bios=bios,
             certificate=certificate,
         )
+
+    def _load_device_instances_from_response(
+        self, device_instance: response_models.PublicDeviceInstance
+    ):
```

Comment:
DeviceInstance is an M2M relationship between Devices and Reports in C3. We need both information about what devices we have and what
From the performance perspective, I've decided to serialise the M2M model instead since it's much easier for Django to retrieve and show the m2m object instead of showing a list of devices and for each of them a list of reports

Reply:
Ah ok I got it. So we use device instance to connect the device to the correct report but we only store the device